rest/route/host_agent_test.go

```diff
@@ -994,6 +1032,65 @@ func TestAssignNextAvailableTaskWithDispatcherSettingsVersionLegacy(t *testing.T
 			So(shouldTeardown, ShouldBeFalse)
 			So(t, ShouldBeNil)
 		})
+		Convey("a task's NumNextTaskDispatches should be increased by 1", func() {
+			taskQueue.Queue = []model.TaskQueueItem{{Id: "stillOkayTask"}}
+			So(taskQueue.Save(), ShouldBeNil)
+			stillOkayTask := task.Task{
+				Id:                    "stillOkayTask",
+				Status:                evergreen.TaskUndispatched,
+				Activated:             true,
+				NumNextTaskDispatches: 4,
+				BuildId:               "b1",
+				Version:               versionId,
+				HostId:                theHostWhoCanBoastTheMostRoast.Id,
+			}
+			So(stillOkayTask.Insert(), ShouldBeNil)
+			testBuild := build.Build{
+				Id:      "b1",
+				Version: versionId,
+			}
+			require.NoError(t, testBuild.Insert())
+
+			testVersion := model.Version{
+				Id: versionId,
+			}
+			require.NoError(t, testVersion.Insert())
+			t, shouldTeardown, err := assignNextAvailableTask(ctx, env, taskQueue, model.NewTaskDispatchService(time.Minute), &theHostWhoCanBoastTheMostRoast, details)
+			So(err, ShouldBeNil)
+			So(shouldTeardown, ShouldBeFalse)
+			So(t, ShouldNotBeNil)
+			taskBFromDb, err := task.FindOneId(stillOkayTask.Id)
+			So(err, ShouldBeNil)
+			So(taskBFromDb, ShouldNotBeNil)
+			So(taskBFromDb.IsStuckTask(), ShouldBeTrue)
+		})
+		Convey("a stuck tasks should not error", func() {
```

Comment:
I don't think this test is relevant anymore it's already covered by the `AStuckNextTaskShouldError` test case, so may be fine to remove.